File: frontend/src/types/column.ts
// ========== Column ==========
export type CategoryCode = 'SP' | 'SC' | 'OVL' | 'DEV' | 'EQP'

export interface ColumnValidation {
  id: number
  rule_type: 'range' | 'required' | 'conditional_required' | 'cross_layer'
  rule_config: Record<string, unknown>
  error_message: string
  is_active: boolean
}

export interface ColumnDefinition {
  id: number
  column_name: string
  display_name: string
  category_id: number
  data_type: 'integer' | 'float' | 'string' | 'select'
  select_options: string[] | null
  unit: string | null
  sort_order: number
  is_required: boolean
  use_yn: boolean
  validations: ColumnValidation[]
}

export interface ColumnCategory {
  id: number
  category_code: CategoryCode
  category_name: string
  sort_order: number
  columns: ColumnDefinition[]
}

File: backend/app/schemas/column.py
from pydantic import BaseModel
from typing import Any


class ColumnValidationResponse(BaseModel):
    id: int
    rule_type: str
    rule_config: dict[str, Any]
    error_message: str
    is_active: bool

    model_config = {"from_attributes": True}


class ColumnDefinitionResponse(BaseModel):
    id: int
    column_name: str
    display_name: str
    category_id: int
    data_type: str
    select_options: list[Any] | None = None
    unit: str | None = None
    sort_order: int
    is_required: bool
    use_yn: bool
    validations: list[ColumnValidationResponse] = []

    model_config = {"from_attributes": True}


class ColumnCategoryResponse(BaseModel):
    id: int
    category_code: str
    category_name: str
    sort_order: int
    columns: list[ColumnDefinitionResponse] = []

    model_config = {"from_attributes": True}
